kotlin-native/backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/descriptors/DescriptorUtils.kt

```diff
@@ -36,8 +40,33 @@ internal val IrClass.implementedInterfaces: List<IrClass>
                 superInterfaces).distinct()
     }
 
-internal val IrFunction.isTypedIntrinsic: Boolean
-    get() = annotations.hasAnnotation(KonanFqNames.typedIntrinsic)
+internal val IrFunction.isTypedIntrinsic: Boolean get() = getTypedIntrinsic() != null
+
+internal fun IrFunction.getTypedIntrinsic(): IrConstructorCall? {
+    if (true) { // @TODO: check laxTypedIntrinsic, do we need to propagate something here, or we can access the context from the Ir node?
```

Comment:
Question: how can I access the LanguageSettings object here? Is it possible to get that instance from an Ir node, or I will need to propagate the instance up to here?